api/v1alpha1/controlplane_types.go

```diff
@@ -71,16 +79,20 @@ const (
 	BackendDBTypeDedicated BackendDBType = "dedicated"
 )
 
-// +kubebuilder:validation:Enum=k8s;ocm;vcluster;host
+// +kubebuilder:validation:Enum=k8s;ocm;vcluster;host;external
 type ControlPlaneType string
 
 const (
 	ControlPlaneTypeK8S      ControlPlaneType = "k8s"
 	ControlPlaneTypeOCM      ControlPlaneType = "ocm"
 	ControlPlaneTypeVCluster ControlPlaneType = "vcluster"
 	ControlPlaneTypeHost     ControlPlaneType = "host"
+	ControlPlaneTypeExternal ControlPlaneType = "external"
 )
 
+// SecretReference is a reference to a secret that holds the kubeconfig for
```

Comment:
It holds two kubeconfigs, in many cases, right?
Where/how do the kubeconfigs appear in the Secret?

Reply:
right. Added clarifications in the comment.